packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -64,30 +65,52 @@ function getVizIndexTypeFromDimensionType(
     }
 }
 
-export class SemanticViewerResultsRunner extends ResultsRunner {
+export class SemanticViewerResultsRunner
+    implements IResultsRunner<VizChartLayout>
+{
     private readonly query: SemanticLayerQuery;
 
     private readonly projectUuid: string;
 
     private readonly fields: SemanticLayerField[];
 
+    private readonly availableFields: SemanticLayerField[];
+
+    private readonly rows: RawResultRow[];
+
+    private readonly dimensions: SemanticLayerField[];
+
+    private readonly metrics: SemanticLayerField[];
+
     constructor({
         query,
         projectUuid,
         fields,
-        ...args
+        rows,
```

Comment:
what's fields vs columns names?

Reply:
I'll make it clearer with some renames. Fields is all of the semantic layer fields. columnNames is just the selected ones.